dev-java/jdom/jdom-2.0.6.1.ebuild

```diff
@@ -40,35 +37,101 @@ RDEPEND="
+
 	# Remove Android stuff to avoid junit RDEPEND.
 	rm -vr contrib/src/java/org/jdom2/contrib/android || die
 
-	# There are resources in JAVA_TEST_SRC_DIR
-	cp -r test/src/{java,resources}/org || die
-	# Remove .java files from JAVA_TEST_RESOURCE_DIRS
-	find test/src/resources -type f -name '*.java' -exec rm -rf {} + || die "deleting classes failed"
+	cd test/src/java || die
+	# java-pkg-simple.eclass expects test resources in JAVA_TEST_RESOURCE_DIRS
+	find . -type f ! -name '*.java' \
+		| xargs cp --parent -t ../resources || die
+
+	sed \
+		-e 's/\r$//g' \
+		-e '/import org.junit.Test/a import org.junit.Ignore;' \
+		-e '/testDocTypeDocument()/i @Ignore' \
+		-i org/jdom2/test/cases/input/TestDOMBuilder.java || die
+
+	sed \
+		-e '/import org.junit.Test/a import org.junit.Ignore;' \
+		-e '/test_TCM__void_setExpandEntities_boolean()/i @Ignore' \
+		-i org/jdom2/test/cases/input/TestSAXBuilder.java || die
+
+	sed \
+		-e 's/\r$//g' \
+		-e '/import org.junit.Test/a import org.junit.Ignore;' \
+		-e '/testDocTypeDocument()/i @Ignore' \
+		-i org/jdom2/test/cases/input/TestStAXEventBuilder.java \
+		-i org/jdom2/test/cases/input/TestStAXStreamBuilder.java || die
+
+	sed \
+		-e '/testWithDocType()/i @Ignore' \
+		-i org/jdom2/test/cases/output/TestDOMOutputter.java || die
+
+	sed \
+		-e '/test_RawSurrogatePair()/i @Ignore' \
+		-e '/test_HighSurrogateAttPair()/i @Ignore' \
+		-e '/test_HighSurrogatePairDecimal()/i @Ignore' \
+		-e '/test_HighSurrogatePair()/i @Ignore' \
+		-e '/test_HighSurrogateAttPairDecimal()/i @Ignore' \
+		-i org/jdom2/test/cases/output/TestStAXEventOutputter.java || die
+
+	sed \
+		-e '/import org.junit.Test/a import org.junit.Ignore;' \
+		-e '/testOutputElementAttributeNotSpecifiedB()/i @Ignore' \
+		-e '/testOutputDocumentOmitEncoding()/i @Ignore' \
+		-e '/testDocTypePublicSystemIDISS()/i @Ignore' \
+		-e '/testDocTypeSimpleISS()/i @Ignore' \
+		-e '/testOutputDocTypeInternalSubset()/i @Ignore' \
+		-e '/testDocTypeSystemIDISS()/i @Ignore' \
+		-i org/jdom2/test/cases/output/AbstractTestOutputter.java || die
+
+	sed \
+		-e '/import org.junit.Test/a import org.junit.Ignore;' \
+		-e '/testNamespaces()/i @Ignore' \
+		-e '/testComplex()/i @Ignore' \
+		-i org/jdom2/test/cases/output/AbstractTestRoundTrip.java || die
+
+	sed \
+		-e '/test_RawSurrogatePair()/i @Ignore' \
+		-e '/test_HighSurrogateAttPair()/i @Ignore' \
+		-e '/test_HighSurrogatePairDecimal()/i @Ignore' \
+		-e '/test_HighSurrogatePair()/i @Ignore' \
+		-e '/test_HighSurrogateAttPairDecimal()/i @Ignore' \
+		-i org/jdom2/test/cases/output/TestStAXStreamOutputter.java || die
+
+	sed \
+		-e '/import org.junit.Test/a import org.junit.Ignore;' \
+		-e '/testFalseUSASCII()/i @Ignore' \
+		-e '/testFalseUTF8()/i @Ignore' \
+		-e '/testFalse()/i @Ignore' \
+		-e '/testValidFalse()/i @Ignore' \
+		-e '/testValidFalseUTF8()/i @Ignore' \
+		-e '/testValidFalseUSASCII()/i @Ignore' \
+		-i org/jdom2/test/cases/special/TestIssue008ExpandEntity.java || die
```

Comment:
why is all this done using `sed` instead of creating a patch?

Reply:
> why is all this done using `sed` instead of creating a patch?

It was mainly because of 3 test classes having `CRLF` line endings leading to very huge patches.
 I've moved all others to a newly created `jdom-2.0.6.1-skip-failing-tests.patch`.